### PR TITLE
Jenkins update and code cleanup

### PR DESCRIPTION
**Pull request by astoltz**
* Update minimum Jenkins version to 1 year old LTS.
* Update packaging dependencies
* General code cleanup
* Update deprecated extensions

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestListenerDecorator.java [hunk 1]
```diff
@@ -26,62 +26,60 @@
...
+    private final Pattern trxPattern = Pattern.compile(TRX_PATTERN);
+    private final Pattern attachmentsPattern = Pattern.compile(ATTACHMENTS_PATTERN);
+    private final Pattern coveragePattern = Pattern.compile(COVERAGE_PATTERN);
 
-    private boolean attachmentsSection;
+    private boolean attachmentsSection = false;
 
-    private String trxFile;
-    private String coverageFile;
+    private String trxFile = null;
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestListenerDecorator.java]: No need to be so verbose it is null by default

File: src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestListenerDecorator.java [hunk 2]
```diff
@@ -26,62 +26,60 @@
...
+    private String coverageFile = null;
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestListenerDecorator.java]: No need to be so verbose it is null by default

File: src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestListenerDecorator.java [hunk 3]
```diff
@@ -26,62 +26,60 @@
...
 
-    private final Pattern trxPattern;
-    private final Pattern attachmentsPattern;
-    private final Pattern coveragePattern;
+    private final Pattern trxPattern = Pattern.compile(TRX_PATTERN);
+    private final Pattern attachmentsPattern = Pattern.compile(ATTACHMENTS_PATTERN);
+    private final Pattern coveragePattern = Pattern.compile(COVERAGE_PATTERN);
 
-    private boolean attachmentsSection;
+    private boolean attachmentsSection = false;
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestListenerDecorator.java]: No need to be so verbose it is false by default

File: src/test/java/org/jenkinsci/plugins/vstest_runner/FileSetTest.java [hunk 4]
```diff
@@ -24,25 +24,16 @@
...
-import hudson.model.Result;
-
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.startsWith;
-
+import hudson.model.*;
```
Comment (casz) [inline src/test/java/org/jenkinsci/plugins/vstest_runner/FileSetTest.java]: No star imports, please.

File: src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestBuilder.java [hunk 5]
```diff
@@ -1,41 +1,24 @@
...
-import hudson.model.AbstractBuild;
-import hudson.model.AbstractProject;
-import hudson.model.Computer;
-import hudson.model.EnvironmentContributingAction;
-import hudson.model.Node;
-import hudson.model.Result;
-import hudson.model.Run;
-import hudson.model.TaskListener;
-import hudson.tasks.Builder;
+import hudson.*;
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestBuilder.java]: No star imports, please.

File: pom.xml [hunk 6]
```diff
@@ -3,12 +3,12 @@
...
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.5</version>
+    <version>3.37</version>
   </parent>
   
   <properties>
-    <jenkins.version>2.7.4</jenkins.version>
-    <java.level>7</java.level>
+    <jenkins.version>2.89.4</jenkins.version>
```
Comment (casz) [inline pom.xml]: why only 2.89.4?
Properly better to go with: 2.107.3 to avoid using `Jenkins.getInstanceOrNull` and just use `Jenkins.get`

Comment (casz) [reply pom.xml]: http://stats.jenkins.io/pluginversions/vstestrunner.html

perhaps I'd suggest just sticking with 2.60.3 just for Java 8 there aren't that many changes if you remove the use of `Jenkins.getInstanceOrNull` and stick to `Jenkins.getInstance`

Comment (astoltz) [reply pom.xml]: I simply looked at what LTS was released a year ago. I'm fine with newer. I'll update the PR.

**Pull request merged**